packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -158,11 +160,14 @@ class _MaterialTextSelectionControls extends TextSelectionControls {
     final double x = (endTextSelectionPoint == null)
         ? startTextSelectionPoint.point.dx
         : (startTextSelectionPoint.point.dx + endTextSelectionPoint.point.dx) / 2.0;
-    final double availableHeight
-        = globalEditableRegion.top - MediaQuery.of(context).padding.top - _kToolbarScreenPadding;
-    final double y = (availableHeight < _kToolbarHeight)
-        ? startTextSelectionPoint.point.dy + globalEditableRegion.height + _kToolbarHeight + _kToolbarScreenPadding
-        : startTextSelectionPoint.point.dy - globalEditableRegion.height;
+    final double toolbarHeightNeeded = MediaQuery.of(context).padding.top
+      + _kToolbarScreenPadding
+      + _kToolbarHeight;
+    final double availableHeight = globalEditableRegion.top + endpoints.first.point.dy - textLineHeight;
+    final bool fitsAbove = toolbarHeightNeeded <= availableHeight;
+    final double y = fitsAbove
```

Comment:
The comment here
https://github.com/flutter/flutter/blob/9be3399fb59347a8e2bfdfd9cd17814456c089f3/packages/flutter/lib/src/material/text_selection.dart#L81-L83
Makes it really confusing. I guess instead of anchor it should say bottom left? And the fact that we use bottom left instead of top left sounds strange to me too.

Reply:
I think the value passed in for position is where the center of the toolbar is drawn? If I pass in `0.0` for y, it draws the toolbar so that it's vertical midpoint is on the top of the textfield.

Reply:
in `_TextSelectionToolbarLayout.getPositionForChild` I think `_TextSelectionToolbarLayout.position` is used as the coordinate of the bottom left vertex of the toolbar, in global coordinate system.

Reply:
Hmm I don't fully understand what that value is meant to be.  Do you think I should update the comment to something specific?  This is what I see when I set `position` to 0,0:

<img width="378" alt="Screen Shot 2019-07-30 at 4 17 16 PM" src="https://user-images.githubusercontent.com/389558/62171927-87f8df80-b2e5-11e9-9be0-5bba384b74b2.png">

Reply:
Ah sorry I was just ranting because it threw me off, It's not really related to this PR.

Reply:
Ok got it.  Yeah it's definitely confusing.